scripts/ping-providers.ts

```diff
@@ -1,21 +1,60 @@
-import { JsonRpcProvider } from 'ethers';
+import { WebClient } from '@slack/web-api';
+import { createPublicClient, http } from 'viem';
 import { CHAINS } from '../src';
 
 const specifiedChain = CHAINS.find((chain) => chain.alias === process.env.CHAIN);
 const chains = specifiedChain ? [specifiedChain] : CHAINS;
 
-chains.forEach(async (chain) => {
-  const provider = new JsonRpcProvider(chain.providerUrl);
-  const chainId = (await provider.getNetwork()).chainId;
-  if (chainId.toString() !== chain.id) {
-    throw new Error(`${chain.alias} provider reports chain ID to be ${chainId}, while it is defined to be ${chain.id}`);
-  }
-  const blockTimestamp = (await provider.getBlock('latest'))!.timestamp;
-  const deltaTime = Math.floor(new Date().getTime() / 1000) - blockTimestamp;
-  const tolerance = 5 * 60;
-  if (Math.abs(deltaTime) > tolerance) {
-    throw new Error(
-      `Timestamp of the latest block the ${chain.alias} provider reports is ${deltaTime} seconds behind the system clock`
-    );
+// Set the following environment variables to also notify to Slack
+const slackToken = process.env.SLACK_TOKEN;
+const slackChannel = process.env.SLACK_CHANNEL;
+const slackClient = new WebClient(slackToken);
+
+async function pingAll(): Promise<PromiseSettledResult<void>[]> {
+  const promises = chains.map(async (chain) => {
+    const client = createPublicClient({ transport: http(chain.providerUrl) });
+
+    const chainId = await client.getChainId();
+    if (chainId.toString() !== chain.id) {
+      throw new Error(`${chain.alias} provider reports chain ID as ${chainId}, while it is defined as ${chain.id}`);
+    }
+
+    const block = await client.getBlock({ blockTag: 'latest' });
+    const blockTimestamp = block.timestamp;
+    const deltaTime = Number(BigInt(Math.floor(new Date().getTime() / 1000)) - blockTimestamp);
+    const tolerance = 5 * 60; // 5 minutes
```

Comment:
Of the ones that fail variably, boba-avalanche has a block time that will be a problem:
"Average block time 11.1 minutes" - https://blockexplorer.avax.boba.network/

But the others the issue looks provider-related given blocks are being produced. What do you think about retrying with a backoff for failed chains?

Reply:
I personally prefer being a bit stricter with the tolerance. Being explicit about the expected tolerance feels a bit better than a hit or miss with an arbitrary (?) 5 minutes. Retrying with a backoff would also significantly extend the time it takes to run the script. Right now it only runs for a few seconds.

We could use the existing `blockTimeMs` value with some some multiplier. e.g. `blockTimeMs * 5`. Alternatively, we could even add a new field specifically for this. e.g. `{ ...chain, "metadata": { "staleBlockToleranceMs": 720000 } }`

boba-avalanche currently has a [`blockTimeMs` value of 70618](https://github.com/api3dao/chains/blob/338b1b700af404a2620c5451bf0c50839f09f742/chains/boba-avalanche.json#L17) (1.1 minutes) That seems a bit off 😅

Reply:
It might still be worth adding retries for the occasional flaky RPC calls

Reply:
> blockTimeMs value with some some multiplier 

This makes a lot of sense, but as you pointed out the issue then becomes keeping these values relatively accurate (even since my last comment boba-avalanche changed from 11.1 to 9.4 min now)

Reply:
Yeah. I'm not sure that there is a good solution here, especially with chains that have such high and unpredictable block times. The quicker, simpler solution is probably the retries with a backoff. What do you think? 5 retries 1 minute apart? Exponential backoff?

Some boba-avalanche blocks are even 20 minutes apart 😞 

https://blockexplorer.avax.boba.network/block/95604/transactions
https://blockexplorer.avax.boba.network/block/95605/transactions

Reply:
> I'm not sure that there is a good solution here, especially with chains that have such high and unpredictable block times. 

Sadly agreed. I think it might be reasonably flake-tolerant with:
- `blockTimeMs * 5` 
- 3 (or 5) retries 1 min apart
- updated `blockTimeMs` of `boba-avalanche`

Reply:
I've implemented the above suggestions but changed the multiplier to 8 instead of 5. Some chains seemed way too close to the tolerance threshold. I think quite a few of our `blockTimeMs` values are not very accurate (I remember Burak raising this concern before).

Reply:
Great solution :+1: 
The problem here is that when you get the block number, some L2s return the transaction number and some return the batch number (and the latter causes these huge block times)